examples/consumefuel/Program.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using Wasmtime;
+
+namespace Example
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            using var engine = new Engine(new Config()
+                .WithFuelConsumption(true));
+            using var module = Module.FromTextFile(engine, "consumefuel.wat");
+            using var linker = new Linker(engine);
+            using var store = new Store(engine);
+
+            linker.Define(
+                "",
+                "expensive",
+                Function.FromCallback(store, (Caller caller) =>
+                {
+                    var remaining = store.ConsumeFuel(1000UL);
```

Comment:
I think we'd want to use `caller` rather than capturing `store`, if possible.

However `Caller` doesn't expose the fuel methods like `Store` does. Do you mind adding `AddFuel`, `ConsumeFuel`, and `GetConsumedFuel` methods to `Caller` (should be identical to those in `Store`)?

That way we can just do:

```c#
var remaining = caller.ConsumeFuel(1000UL);
```

And the callback has no captured variables.

Reply:
Agreed - much better. Thanks!

also snuck in one last change - I added the example to CI (didn't see the build step that runs the examples until now)